src/app/public/modules/text-editor/services/text-editor.service.ts

```diff
@@ -0,0 +1,501 @@
+        color: #686c73;
+        font-weight: 400;
+        font-size: 15px;
+        font-style: italic;
+      }`;
+      documentEl.head.appendChild(styleEl);
+
+      const style: SkyTextEditorStyleState = { ...STYLE_STATE_DEFAULTS, ...styleState };
+      const bodyStyle = `background-color: ${style.backColor};
+        color: ${style.fontColor};
+        font-family: ${style.font};
+        font-size: ${style.fontSize}px`;
+      documentEl.querySelector('html').setAttribute('lang', 'en');
+      documentEl.body.setAttribute('contenteditable', 'true');
+      documentEl.body.setAttribute('id', id);
+      documentEl.body.setAttribute('role', 'main');
+      documentEl.body.setAttribute('class', 'editor');
+      documentEl.body.setAttribute('style', bodyStyle);
+      documentEl.body.setAttribute('data-placeholder', placeholder || '');
+    }
+  }
+
+  public setPlaceholder(id: string, placeholder?: string): void {
+    const documentEl = this.getDocumentEl(id);
+    documentEl.body.setAttribute('data-placeholder', placeholder || '');
+  }
+
+  public removeEditor(id: string): void {
+    if (id in this.editors) {
+      SkyTextEditorService.removeObservers(this.editors[id]);
+      delete this.editors[id];
+    }
+  }
+
+  public execCommand(id: string, editorCommand: EditorCommand): void {
+    if (id in this.editors) {
+      const documentEl = this.getDocumentEl(id);
+
+      if (!this.editorSelected(documentEl)) {
+        this.focusEditor(id);
+      }
+
+      if (this.editorSelected(documentEl)) {
+        const commandIsSupportedAndEnabled = documentEl.execCommand(editorCommand.command, false, editorCommand.value);
+
+        // IE11 doesn't support insertHTML
+        /* istanbul ignore next */
+        if (
+          editorCommand.command.toLowerCase() === 'inserthtml' &&
+          !commandIsSupportedAndEnabled
+        ) {
+          this.insertHtmlInIE11(id, editorCommand.value);
+        }
+
+        /* istanbul ignore next */
+        if (!this.editorSelected(documentEl)) {
+          this.focusEditor(id);
+        }
+        this.editors[id].commandChangeObservable.next();
+      }
+    }
+  }
+
+  public setFontSize(id: string, fontSize: number): void {
+    const doc = this.getDocumentEl(id);
+    this.execCommand(id, {command: 'fontSize', value: 1});
+    const fontElements: HTMLElement[] = Array.from(doc.querySelectorAll('font[size="1"]'));
+    for (let element of fontElements) {
+      element.removeAttribute('size');
+      element.style.fontSize = (fontSize + 'px');
+    }
+    SkyTextEditorService.cleanUpBlankStyleTags(doc);
+
+    /* istanbul ignore next */
+    if (!this.editorSelected(doc)) {
+      this.focusEditor(id);
+    }
+    this.editors[id].commandChangeObservable.next();
+  }
+
+  public selectionChangeListener(id: string): Observable<unknown> {
+    return this.editors[id].selectionChangeObservable;
+  }
+
+  public clickListener(id: string): Observable<unknown> {
```

Comment:
These methods in this section seem a little vague to me. For example, on this one I wouldn't know if `clickListener` is clicking some sort of listener or getting some sort of click listener. (I know that in this case it is the later but only because I read the source code).

Reply:
I think this particular one is based on it being used as the `listener` argument in the `.addEventListener()` method inside this same service (line 355). Do you have any suggestions on how we could improve the name?

Reply:
I'm not going to lose sleep over it - but I think something even as simple as `getClickListener` would make this a little clearer. `click` being both a noun and a verb is confusing :)

Reply:
I'm probably overthinking this though. I'm fine if we leave it.